src/main.cc

```diff
@@ -13,18 +13,24 @@ int main(int argc, char** argv) {
     //! IO
     std::unique_ptr<IO> io(new IO(argc, argv));
 
-    //! Mesh
-    std::unique_ptr<Mesh<3, 8>> mesh(new GMSH<3, 8>());
-
     //! MaterialProperties
     std::shared_ptr<MaterialProperties> material =
         std::make_shared<MaterialProperties>(io->material_properties());
 
+    // const unsigned Tdim = io->dimension();
```

Comment:
I can't seem to use this. Can you let me know how to cast from `non-const` to `const`? Thanks!

Reply:
You need a `switch` to handle dimensions

Reply:
Yes but we need to initialize `const` variable and `switch` would not allow cross-initialization. Or maybe I don't get the way you want to do it?